src/Analyzers/Core/Analyzers/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs

```diff
@@ -16,368 +16,411 @@
-                return;
-
-            if (!containingType.Equals(property.ContainingType))
-                return;
-
-            if (property.IsIndexer)
-                return;
-
-            // The property can't be virtual.  We don't know if it is overridden somewhere.  If it 
-            // is, then calls to it may not actually assign to the field.
-            if (property.IsVirtual || property.IsOverride || property.IsSealed)
-                return;
-
-            if (property.IsWithEvents)
-                return;
-
-            if (property.Parameters.Length > 0)
-                return;
-
-            // Need at least a getter.
-            if (property.GetMethod == null)
-                return;
-
-            if (!CanExplicitInterfaceImplementationsBeFixed() && property.ExplicitInterfaceImplementations.Length != 0)
-                return;
-
-            // Serializable types can depend on fields (and their order).  Don't report these
-            // properties in that case.
-            if (containingType.IsSerializable)
-                return;
-
-            var preferAutoProps = context.GetAnalyzerOptions().PreferAutoProperties;
-            if (!preferAutoProps.Value)
-                return;
-
-            // Avoid reporting diagnostics when the feature is disabled. This primarily avoids reporting the hidden
-            // helper diagnostic which is not otherwise influenced by the severity settings.
-            var severity = preferAutoProps.Notification.Severity;
-            if (severity == ReportDiagnostic.Suppress)
-                return;
-
-            var getterField = GetGetterField(semanticModel, property.GetMethod, cancellationToken);
-            if (getterField == null)
-                return;
-
-            // Only support this for private fields.  It limits the scope of hte program
-            // we have to analyze to make sure this is safe to do.
-            if (getterField.DeclaredAccessibility != Accessibility.Private)
-                return;
 
-            // If the user made the field readonly, we only want to convert it to a property if we
-            // can keep it readonly.
-            if (getterField.IsReadOnly && !SupportsReadOnlyProperties(compilation))
-                return;
-
-            // Field and property have to be in the same type.
-            if (!containingType.Equals(getterField.ContainingType))
-                return;
+                return true;
+            }
+        }, SymbolKind.NamedType);
+
+    private void RegisterNonConstructorFieldWrites(
+        HashSet<string> fieldNames,
+        ConcurrentDictionary<IFieldSymbol, ConcurrentSet<SyntaxNode>> fieldWrites,
+        SemanticModel semanticModel,
+        SyntaxNode codeBlock,
+        CancellationToken cancellationToken)
+    {
+        if (codeBlock.FirstAncestorOrSelf<TConstructorDeclaration>() != null)
```

Comment:
this logic moved up from VB to the base class.  it's needed by both c# and vb now.